Added Tips and Tricks Document (#190)
o3de/community: process-docs-and-markdown/discord-tips-tricks-troubleshooting.md

# Discord Tips, Tricks and Troubleshooting

> - [Discord Tips, Tricks and Troubleshooting](#discord-tips-tricks-and-troubleshooting)
>   * [Two-Factor Authentication (2FA)](#two-factor-authentication-2fa)
>   * [Screen Sharing](#screen-sharing)
>   * [Change Username on Discord](#change-username-on-discord)
>   * [Custom Discord Rich Presence](#custom-discord-rich-presence)
>   * [Text Formatting](#text-formatting)
>   * [Color Codes](#color-codes)
>   * [Privacy and Notifications](#privacy-and-notifications)
>     + [Privacy](#privacy)
>       - [Desktop Privacy Settings:](#desktop-privacy-settings)
>       - [Mobile Privacy Settings:](#mobile-privacy-settings)
>     + [Notifications](#notifications)
>       - [Desktop Notification Settings:](#desktop-notification-settings)
>       - [Mobile Notification Settings:](#mobile-notification-settings)
>   * [Error messages](#error-messages)

Reviewer: Remove the H1 from the table of contents this section is in H1 already
```suggestion
- [Two-Factor Authentication (2FA)](#two-factor-authentication-2fa)
- [Screen Sharing](#screen-sharing)
- [Change Username on Discord](#change-username-on-discord)
- [Custom Discord Rich Presence](#custom-discord-rich-presence)
- [Text Formatting](#text-formatting)
- [Color Codes](#color-codes)
- [Privacy and Notifications](#privacy-and-notifications)
- [Privacy](#privacy)
    - [Desktop Privacy Settings:](#desktop-privacy-settings)
    - [Mobile Privacy Settings:](#mobile-privacy-settings)
- [Notifications](#notifications)
    - [Desktop Notification Settings:](#desktop-notification-settings)
    - [Mobile Notification Settings:](#mobile-notification-settings)
- [Error messages](#error-messages)
```

5. You will then be prompted to choose a verification method: either using an authentication app or receiving a text message with a verification code. If you choose an authentication app, you will need to download an authenticator app (such as Google Authenticator or Authy) on your mobile device and scan the QR code displayed on the screen. If you choose to receive a text message, you will need to enter your phone number and wait for the code to be sent to you.
6. After verifying your account with either the authentication app or the text message, Discord will generate backup codes that you can use to access your account in case you lose access to your verification method. Be sure to store these codes in a safe place!

That's it! Your account should now be protected with two-factor authentication. From now on, when you log in to Discord on a new device or browser, you will need to enter a verification code from your authentication app or from a text message before gaining access to your account.

https://support.discord.com/hc/en-us/articles/219576828-Setting-up-Two-Factor-Authentication

## Screen Sharing

> To enable streamin on Discord, you will need to follow these steps:

Reviewer: ```suggestion
To enable streaming on Discord, you will need to follow these steps:
```

For example, if your current username is AMZN_AmyFinch and you want to change it to adhere to Option A, you would change it to "Amy [Amazon]". If you want to use Option B, you could change it to "Finchy [Developer]". If you want to use Option C, you could change it to "Amy".

Here are the step-by-step instructions to change your username:

1.  Open the Discord app or website and log in to your account.
2.  On the left-hand side, click on the server where you want to change your username.
3.  Find your username in the member list or the chat window, and click on it.
4.  Click on the three dots on the right side of your username, then select "Change Nickname."
5.  Type in your new username according to the guidelines provided above.
6.  Click "Save" to update your username.

Please note that some servers may have specific guidelines for usernames, including limitations on the number of characters or how often you can change your username. Be sure to check the server guidelines or contact an administrator if you have any questions or issues.

For reference, here are some examples of usernames that adhere to the recommended format:

> *  Royal OBrien [Amazon]

Reviewer: I know this is only an example, but I think it's better to avoid making an incorrect association 
```suggestion
*  Royal OBrien [LF]
```

*  Finchy [Finch Studio]
*  OBWANDO [Developer]
*  Finchy

> https://support.discord.com/hc/en-us/articles/213480948-How-do-I-change-my-Username-
> https://support.discord.com/hc/en-us/articles/219070107-Server-Nicknames 

Reviewer: ```suggestion
For more information, refer to Discord's articles, [How do I change my Username?](https://support.discord.com/hc/en-us/articles/213480948-How-do-I-change-my-Username-) and [Server Nicknames](https://support.discord.com/hc/en-us/articles/219070107-Server-Nicknames).
```

1. Open the config.json file and replace the placeholders with your Discord application's client ID and the custom status message and details you want to display.
1. To get your Discord application's client ID, go to the Discord Developer Portal (https://discord.com/developers/applications) and create a new application or select an existing one. Go to the "General Information" section and copy the client ID.
1. Customize the status message and details in the config.json file. You can use the placeholders provided or enter your own text.
1. Save the config.json file.

**Running the script:**

1. Open the command prompt or terminal and navigate to the downloaded folder.
1. Run node index.js to start the script.
1. Your custom Discord status should now be displayed on your profile.

Note: Keep the command prompt or terminal open while you want the status to be displayed. If you close it, the status will disappear. Also, make sure to update the config.json file whenever you want to change your status message or details.

Alternativly, visit:
https://docs.customrp.xyz/setting-up

## Text Formatting

Here's a comprehensive list of all the formatting options available in Discord:

> 1. Bold text: To make text bold, surround it with two asterisks on both sides like this:<br>

Reviewer: ```suggestion
1. **Bold text**: To make text bold, surround it with two asterisks on both sides like this:<br>
```

    `**Bold Text**`

> 2. *Italicized text:* To italicize text, surround it with a single asterisk on both sides like this:<br>

Reviewer: ```suggestion
2. *Italicized text*: To italicize text, surround it with a single asterisk on both sides like this:<br>
```

    `*italicized text*`
3. ***Italicized Bold text***: To italicize bold text, surround it with three asterisk on both sides like this:<br> 
    `***italicized bold text***`

> 4. Underlined text: To underline text, surround it with two underscores on both sides like this:<br>
>     `**Underlined Text**`

Reviewer: ```suggestion
4. Underlined text: To underline text, surround it with two underscores on both sides like this:<br>
    `_Underlined Text_`
```

5. *_Underlined and Italicized Text:_ *To underline and italicize text, add two underscores along with an asterisk on both sides like this:<br>
    `__*Underlined and Italicized Text*__`

> 6. **_Underlined and Bolded Text_**: To underline and bold text, add two underscores along with two asterisks on both sides like this:<br>
>     `__**Underlined and Bolded Text**__`

Reviewer: ```suggestion
6. **_Underlined and bold text_**: To underline and bold text, add two underscores along with two asterisks on both sides like this:<br>
    `__**Underlined and bold text**__`
```

> 7. ***_Underlined, italicize and Bolded Text_***: To underline and bold text, add two underscores along with three asterisks on both sides like this:<br>
>     `__***Underlined, italicize and Bolded Text***__`

Reviewer: ```suggestion
7. ***_Underlined, italicize and bold text_***: To underline and bold text, add two underscores along with three asterisks on both sides like this:<br>
    `__***Underlined, italicize and bold text***__`
```

8. __***I want this to be shown in its full glory!***___

>     In case you actually *want* to see your underscores or asterisks in a message (like in an emoji, for example), you can use the backslash ‘\’ key to skip markdown formatting and show the text just like it is:<br> 

Reviewer: ```suggestion
    In case you actually *want* to see your underscores or asterisks in a message (like in an emoji, for example), you can use the backslash `\` key to skip markdown formatting and show the text just like it is:<br> 
```

    `\_\_\*\*\*I want this to be shown in its full glory!\*\*\*\_\_\_`
10. Strikethrough text: To add a strikethrough to text, surround it with two tildes on either side, like this: 
    `~~Strikethrough Text~~`

> 11. Formatting on International Keyboard Layouts
>     **German:** Shift+[+] (key right of Ü)
>     **Spanish:** Shift+[+] (key right of `^ (Spain) or ´¨ (Latin America))
>     **French (France):** * (key right of ù%)
>     **French (Belgium):** Shift+$ (key right of ^¨)
>     **French (Switzerland):** Shift+3
>     **Italian:** Shift+[+] (key right of èé)
>     **Swedish:** Shift+’ (key right of Ä) 

Reviewer: There are some formatting issues as shown: 
![image](https://user-images.githubusercontent.com/75444793/229642257-e2ca514b-82e4-445c-b3fc-7183cea3735a.png)

```suggestion
11. Formatting on International Keyboard Layouts
   - **German:** Shift+[+] (key right of Ü)
   - **Spanish:** Shift+[+] (key right of `^ (Spain) or ´¨ (Latin America))
   - **French (France):** * (key right of ù%)
   - **French (Belgium):** Shift+$ (key right of ^¨)
   - **French (Switzerland):** Shift+3
   - **Italian:** Shift+[+] (key right of èé)
   - **Swedish:** Shift+’ (key right of Ä) 
```

1. Here you can customize various notification settings such as:
      1. Push Notifications: You can choose to receive push notifications for new messages and mentions. You can also choose how frequently you want to receive push notifications.
      1. In-App Notifications: You can choose whether or not to receive notifications for new messages and mentions while you're actively using the app.
      1. Sound and Vibration: You can choose the sound that plays when you receive a notification, as well as whether or not your device vibrates when you receive a notification.
      1. Server Notification Settings: You can choose which notifications you want to receive for each server you're a member of, such as @mentions, direct messages, or server updates.
1. Once you've adjusted your settings, be sure to tap on the Save button in the top-right corner of the screen.

By customizing your notification settings on both desktop and mobile versions of Discord, you can ensure that you only receive notifications for the messages and mentions that are most important to you, and that you receive them in a way that's convenient for you.

## Error messages

> 1. When Creating an event on discord and shown with "Missing Permissions", ensure that you have 2FA enabled. 
> 
> 2. If you message a member on discord and represented with a message saying it could not be sent like so:

Reviewer: ```suggestion
1. When Creating an event on Discord and shown with "Missing Permissions", ensure that you have 2FA enabled. 

2. If you message a member on Discord and represented with a message saying it could not be sent like so:
```

```
Your message could not be delivered. This is usually because you don't share a server with the recipient or the recipient is only accepting direct messages from friends. You can see the full list of reasons here:
``` 

> This means that you have not added them as a friend. This is normally a setting thats enabled on the other users end to prevent spam. 

Reviewer: ```suggestion
This means that you have not added them as a friend. This is normally a setting that's enabled on the other users end to prevent spam. 
```

> Note, If you have any issues with any of these, ping me on discord Finchy#5545

Reviewer: As a long term solution, you may consider changing this to refer to the Discord Help Center.
```suggestion
Note, if you have any issues or further questions, refer to the [Discord Help Center](https://support.discord.com/hc/en-us).
```
